apps/server/test/lexical/server/code_intelligence/completion/env_test.exs

```diff
@@ -228,12 +233,132 @@ defmodule Lexical.Server.CodeIntelligence.Completion.EnvTest do
     end
   end
 
+  describe "in_context?(env, :struct_arguments)" do
+    def wrap_with_module(text) do
+      """
+      defmodule MyModule do
+        #{text}
+      end
+      """
+    end
+
+    def wrap_with_function(text) do
+      """
+      def func do
+        #{text}
+      end
+      """
+    end
+
+    def wrap_with_function_arguments(text) do
+      """
+      def func(#{text}) do
+      end
+      """
+    end
+
+    test "is true if the cursor is directly after the opening curly" do
+      env = "%User{|}" |> wrap_with_module() |> new_env()
+      assert in_context?(env, :struct_arguments)
+    end
+
+    test "is true when the struct is in the function variable" do
+      env = "%User{|}" |> wrap_with_function() |> wrap_with_module() |> new_env()
+      assert in_context?(env, :struct_arguments)
+    end
+
+    test "is true when the struct is in the function arguments" do
+      env = "%User{|}" |> wrap_with_function_arguments() |> wrap_with_module() |> new_env()
+      assert in_context?(env, :struct_arguments)
+    end
+
+    test "is false when the cursor is not in a container" do
+      env = new_env("%User{|}")
+      refute in_context?(env, :struct_arguments)
+    end
+
+    test "is true if the cursor is after the field name" do
+      env = "%User{name: |}" |> wrap_with_module() |> new_env()
+      assert in_context?(env, :struct_arguments)
+    end
+
+    test "is true if the cursor is after the field value" do
+      env = "%User{name: \"John\"|}" |> wrap_with_module() |> new_env()
+      assert in_context?(env, :struct_arguments)
+    end
+
+    test "is true if the cursor starts in the middle of the struct" do
+      env = "%User{name: \"John\", |}" |> wrap_with_module() |> new_env()
+      assert in_context?(env, :struct_arguments)
+    end
+
+    test "is false if the cursor is after the closing curly" do
+      env = "%User{}|" |> wrap_with_module() |> new_env()
+      refute in_context?(env, :struct_arguments)
+    end
+
```

Comment:
```elixir
test "is false in static keywords" do 
  env = "[foo: |]" |> wrap_with_module() |> new_env() 
  refute in_context?(env, :struct_arguments)
end
```

Also add the same test with function arguments and another for map keys.
Though now that I think of this, what you're really checking for isn't struct arguments, it's map update syntax.